### ISSUE TITLE
QGIS 2.18.14 extent reload does not run

### ISSUE THREAD

**Issue by klokan** [geometalab/Vector-Tiles-Reader-QGIS-Plugin#175]
On Mac with QGIS 2.18.14 and latest stable version of this plugin -  the base map does not reload data on panning with the default base map.

In the lower left corner is still zoom 0 - Tile 0,0,0
I am surprised that EPSG:3857 (OTF) - so with On the fly transformation is enabled.

There are no visible error logs under "Vector Tile Reader" log. Native decoding is supported.

I expect the problem is:

```
Reloading due to scale change from 'None' (zoom None) to '350641802734' (zoom 0)
```

Log:
```
2018-01-18T08:45:38	0	Vector Tiles Reader Plugin loaded...
2018-01-18T08:46:53	0	Bounds of source: {'width': 1, 'scheme': 'xyz', 'y_min': 0, 'x_max': 0, 'x_min': 0, 'y_max': 0, 'zoom': 0, 'height': 1}
2018-01-18T08:46:53	0	Loading zoom level '0', bounds: {'width': 1, 'scheme': 'xyz', 'y_min': 0, 'x_max': 0, 'x_min': 0, 'y_max': 0, 'zoom': 0, 'height': 1}
2018-01-18T08:46:53	0	Native decoding supported!!!
2018-01-18T08:46:53	0	Tile limit enabled: True
2018-01-18T08:46:53	0	1 tiles in cache. Max. 0 will be loaded additionally.
2018-01-18T08:46:53	0	Processing features of 1 tiles...
2018-01-18T08:46:53	0	Creating layers...
2018-01-18T08:46:54	0	Refresh layers...
2018-01-18T08:46:54	0	Adding new layers...
2018-01-18T08:46:54	0	Styling layers...
2018-01-18T08:46:54	0	Import complete
2018-01-18T08:46:54	0	Loading of zoom level 0 complete! Loaded extent: {'width': 1, 'scheme': 'xyz', 'y_min': 0, 'x_max': 0, 'x_min': 0, 'y_max': 0, 'zoom': 0, 'height': 1}
2018-01-18T08:48:31	0	Vector Tiles Reader Plugin loaded...
2018-01-18T08:48:31	0	Reloading due to scale change from 'None' (zoom None) to '350641802734' (zoom 0)
2018-01-18T08:48:31	0	Loading zoom level '0', bounds: {'width': 1, 'scheme': 'xyz', 'y_min': 0, 'x_max': 0, 'x_min': 0, 'y_max': 0, 'zoom': 0, 'height': 1}
```
Python warnings:
```
2018-01-18T08:45:39	1	warning:/Library/Frameworks/SQLite3.framework/Versions/C/Python/2.7/pyspatialite/dbapi2.py:81: DeprecationWarning: Converters and adapters are deprecated. Please use only supported SQLite types. Any type mapping should happen in layer above this module.
			  register_adapter(datetime.date, adapt_date)
			
			traceback:  File "<string>", line 1, in <module>
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/qgis/utils.py", line 333, in startPlugin
			    plugins[packageName] = package.classFactory(iface)
			  File "/Users/klokan/.qgis2/python/plugins/qgis2web/__init__.py", line 28, in classFactory
			    from .qgis2web import Qgis2Web
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/qgis/utils.py", line 607, in _import
			    mod = _builtin_import(name, globals, locals, fromlist, level)
			  File "/Users/klokan/.qgis2/python/plugins/qgis2web/qgis2web.py", line 25, in <module>
			    from processing.core.Processing import Processing
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/qgis/utils.py", line 607, in _import
			    mod = _builtin_import(name, globals, locals, fromlist, level)
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/plugins/processing/__init__.py", line 29, in <module>
			    from processing.tools.general import *              # NOQA
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/qgis/utils.py", line 607, in _import
			    mod = _builtin_import(name, globals, locals, fromlist, level)
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/plugins/processing/tools/general.py", line 31, in <module>
			    from processing.core.Processing import Processing
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/qgis/utils.py", line 607, in _import
			    mod = _builtin_import(name, globals, locals, fromlist, level)
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/plugins/processing/core/Processing.py", line 42, in <module>
			    from processing.script.ScriptUtils import ScriptUtils
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/qgis/utils.py", line 607, in _import
			    mod = _builtin_import(name, globals, locals, fromlist, level)
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/plugins/processing/script/ScriptUtils.py", line 30, in <module>
			    from processing.script.ScriptAlgorithm import ScriptAlgorithm
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/qgis/utils.py", line 607, in _import
			    mod = _builtin_import(name, globals, locals, fromlist, level)
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/plugins/processing/script/ScriptAlgorithm.py", line 33, in <module>
			    from processing.core.GeoAlgorithm import GeoAlgorithm
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/qgis/utils.py", line 607, in _import
			    mod = _builtin_import(name, globals, locals, fromlist, level)
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/plugins/processing/core/GeoAlgorithm.py", line 41, in <module>
			    from processing.core.parameters import ParameterRaster, ParameterVector, ParameterMultipleInput, ParameterTable, Parameter
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/qgis/utils.py", line 607, in _import
			    mod = _builtin_import(name, globals, locals, fromlist, level)
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/plugins/processing/core/parameters.py", line 32, in <module>
			    from processing.tools.vector import resolveFieldIndex
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/qgis/utils.py", line 607, in _import
			    mod = _builtin_import(name, globals, locals, fromlist, level)
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/plugins/processing/tools/vector.py", line 47, in <module>
			    from processing.tools import dataobjects, spatialite, postgis
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/qgis/utils.py", line 607, in _import
			    mod = _builtin_import(name, globals, locals, fromlist, level)
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/plugins/processing/tools/spatialite.py", line 29, in <module>
			    from pyspatialite import dbapi2 as sqlite
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/qgis/utils.py", line 607, in _import
			    mod = _builtin_import(name, globals, locals, fromlist, level)
			  File "/Library/Frameworks/SQLite3.framework/Versions/C/Python/2.7/pyspatialite/dbapi2.py", line 86, in <module>
			    register_adapters_and_converters()
			
2018-01-18T08:45:39	1	warning:/Library/Frameworks/SQLite3.framework/Versions/C/Python/2.7/pyspatialite/dbapi2.py:82: DeprecationWarning: Converters and adapters are deprecated. Please use only supported SQLite types. Any type mapping should happen in layer above this module.
			  register_adapter(datetime.datetime, adapt_datetime)
			
			traceback:  File "<string>", line 1, in <module>
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/qgis/utils.py", line 333, in startPlugin
			    plugins[packageName] = package.classFactory(iface)
			  File "/Users/klokan/.qgis2/python/plugins/qgis2web/__init__.py", line 28, in classFactory
			    from .qgis2web import Qgis2Web
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/qgis/utils.py", line 607, in _import
			    mod = _builtin_import(name, globals, locals, fromlist, level)
			  File "/Users/klokan/.qgis2/python/plugins/qgis2web/qgis2web.py", line 25, in <module>
			    from processing.core.Processing import Processing
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/qgis/utils.py", line 607, in _import
			    mod = _builtin_import(name, globals, locals, fromlist, level)
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/plugins/processing/__init__.py", line 29, in <module>
			    from processing.tools.general import *              # NOQA
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/qgis/utils.py", line 607, in _import
			    mod = _builtin_import(name, globals, locals, fromlist, level)
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/plugins/processing/tools/general.py", line 31, in <module>
			    from processing.core.Processing import Processing
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/qgis/utils.py", line 607, in _import
			    mod = _builtin_import(name, globals, locals, fromlist, level)
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/plugins/processing/core/Processing.py", line 42, in <module>
			    from processing.script.ScriptUtils import ScriptUtils
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/qgis/utils.py", line 607, in _import
			    mod = _builtin_import(name, globals, locals, fromlist, level)
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/plugins/processing/script/ScriptUtils.py", line 30, in <module>
			    from processing.script.ScriptAlgorithm import ScriptAlgorithm
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/qgis/utils.py", line 607, in _import
			    mod = _builtin_import(name, globals, locals, fromlist, level)
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/plugins/processing/script/ScriptAlgorithm.py", line 33, in <module>
			    from processing.core.GeoAlgorithm import GeoAlgorithm
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/qgis/utils.py", line 607, in _import
			    mod = _builtin_import(name, globals, locals, fromlist, level)
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/plugins/processing/core/GeoAlgorithm.py", line 41, in <module>
			    from processing.core.parameters import ParameterRaster, ParameterVector, ParameterMultipleInput, ParameterTable, Parameter
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/qgis/utils.py", line 607, in _import
			    mod = _builtin_import(name, globals, locals, fromlist, level)
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/plugins/processing/core/parameters.py", line 32, in <module>
			    from processing.tools.vector import resolveFieldIndex
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/qgis/utils.py", line 607, in _import
			    mod = _builtin_import(name, globals, locals, fromlist, level)
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/plugins/processing/tools/vector.py", line 47, in <module>
			    from processing.tools import dataobjects, spatialite, postgis
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/qgis/utils.py", line 607, in _import
			    mod = _builtin_import(name, globals, locals, fromlist, level)
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/plugins/processing/tools/spatialite.py", line 29, in <module>
			    from pyspatialite import dbapi2 as sqlite
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/qgis/utils.py", line 607, in _import
			    mod = _builtin_import(name, globals, locals, fromlist, level)
			  File "/Library/Frameworks/SQLite3.framework/Versions/C/Python/2.7/pyspatialite/dbapi2.py", line 86, in <module>
			    register_adapters_and_converters()
			
2018-01-18T08:45:39	1	warning:/Library/Frameworks/SQLite3.framework/Versions/C/Python/2.7/pyspatialite/dbapi2.py:83: DeprecationWarning: Converters and adapters are deprecated. Please use only supported SQLite types. Any type mapping should happen in layer above this module.
			  register_converter("date", convert_date)
			
			traceback:  File "<string>", line 1, in <module>
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/qgis/utils.py", line 333, in startPlugin
			    plugins[packageName] = package.classFactory(iface)
			  File "/Users/klokan/.qgis2/python/plugins/qgis2web/__init__.py", line 28, in classFactory
			    from .qgis2web import Qgis2Web
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/qgis/utils.py", line 607, in _import
			    mod = _builtin_import(name, globals, locals, fromlist, level)
			  File "/Users/klokan/.qgis2/python/plugins/qgis2web/qgis2web.py", line 25, in <module>
			    from processing.core.Processing import Processing
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/qgis/utils.py", line 607, in _import
			    mod = _builtin_import(name, globals, locals, fromlist, level)
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/plugins/processing/__init__.py", line 29, in <module>
			    from processing.tools.general import *              # NOQA
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/qgis/utils.py", line 607, in _import
			    mod = _builtin_import(name, globals, locals, fromlist, level)
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/plugins/processing/tools/general.py", line 31, in <module>
			    from processing.core.Processing import Processing
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/qgis/utils.py", line 607, in _import
			    mod = _builtin_import(name, globals, locals, fromlist, level)
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/plugins/processing/core/Processing.py", line 42, in <module>
			    from processing.script.ScriptUtils import ScriptUtils
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/qgis/utils.py", line 607, in _import
			    mod = _builtin_import(name, globals, locals, fromlist, level)
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/plugins/processing/script/ScriptUtils.py", line 30, in <module>
			    from processing.script.ScriptAlgorithm import ScriptAlgorithm
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/qgis/utils.py", line 607, in _import
			    mod = _builtin_import(name, globals, locals, fromlist, level)
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/plugins/processing/script/ScriptAlgorithm.py", line 33, in <module>
			    from processing.core.GeoAlgorithm import GeoAlgorithm
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/qgis/utils.py", line 607, in _import
			    mod = _builtin_import(name, globals, locals, fromlist, level)
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/plugins/processing/core/GeoAlgorithm.py", line 41, in <module>
			    from processing.core.parameters import ParameterRaster, ParameterVector, ParameterMultipleInput, ParameterTable, Parameter
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/qgis/utils.py", line 607, in _import
			    mod = _builtin_import(name, globals, locals, fromlist, level)
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/plugins/processing/core/parameters.py", line 32, in <module>
			    from processing.tools.vector import resolveFieldIndex
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/qgis/utils.py", line 607, in _import
			    mod = _builtin_import(name, globals, locals, fromlist, level)
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/plugins/processing/tools/vector.py", line 47, in <module>
			    from processing.tools import dataobjects, spatialite, postgis
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/qgis/utils.py", line 607, in _import
			    mod = _builtin_import(name, globals, locals, fromlist, level)
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/plugins/processing/tools/spatialite.py", line 29, in <module>
			    from pyspatialite import dbapi2 as sqlite
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/qgis/utils.py", line 607, in _import
			    mod = _builtin_import(name, globals, locals, fromlist, level)
			  File "/Library/Frameworks/SQLite3.framework/Versions/C/Python/2.7/pyspatialite/dbapi2.py", line 86, in <module>
			    register_adapters_and_converters()
			
2018-01-18T08:45:39	1	warning:/Library/Frameworks/SQLite3.framework/Versions/C/Python/2.7/pyspatialite/dbapi2.py:84: DeprecationWarning: Converters and adapters are deprecated. Please use only supported SQLite types. Any type mapping should happen in layer above this module.
			  register_converter("timestamp", convert_timestamp)
			
			traceback:  File "<string>", line 1, in <module>
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/qgis/utils.py", line 333, in startPlugin
			    plugins[packageName] = package.classFactory(iface)
			  File "/Users/klokan/.qgis2/python/plugins/qgis2web/__init__.py", line 28, in classFactory
			    from .qgis2web import Qgis2Web
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/qgis/utils.py", line 607, in _import
			    mod = _builtin_import(name, globals, locals, fromlist, level)
			  File "/Users/klokan/.qgis2/python/plugins/qgis2web/qgis2web.py", line 25, in <module>
			    from processing.core.Processing import Processing
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/qgis/utils.py", line 607, in _import
			    mod = _builtin_import(name, globals, locals, fromlist, level)
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/plugins/processing/__init__.py", line 29, in <module>
			    from processing.tools.general import *              # NOQA
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/qgis/utils.py", line 607, in _import
			    mod = _builtin_import(name, globals, locals, fromlist, level)
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/plugins/processing/tools/general.py", line 31, in <module>
			    from processing.core.Processing import Processing
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/qgis/utils.py", line 607, in _import
			    mod = _builtin_import(name, globals, locals, fromlist, level)
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/plugins/processing/core/Processing.py", line 42, in <module>
			    from processing.script.ScriptUtils import ScriptUtils
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/qgis/utils.py", line 607, in _import
			    mod = _builtin_import(name, globals, locals, fromlist, level)
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/plugins/processing/script/ScriptUtils.py", line 30, in <module>
			    from processing.script.ScriptAlgorithm import ScriptAlgorithm
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/qgis/utils.py", line 607, in _import
			    mod = _builtin_import(name, globals, locals, fromlist, level)
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/plugins/processing/script/ScriptAlgorithm.py", line 33, in <module>
			    from processing.core.GeoAlgorithm import GeoAlgorithm
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/qgis/utils.py", line 607, in _import
			    mod = _builtin_import(name, globals, locals, fromlist, level)
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/plugins/processing/core/GeoAlgorithm.py", line 41, in <module>
			    from processing.core.parameters import ParameterRaster, ParameterVector, ParameterMultipleInput, ParameterTable, Parameter
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/qgis/utils.py", line 607, in _import
			    mod = _builtin_import(name, globals, locals, fromlist, level)
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/plugins/processing/core/parameters.py", line 32, in <module>
			    from processing.tools.vector import resolveFieldIndex
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/qgis/utils.py", line 607, in _import
			    mod = _builtin_import(name, globals, locals, fromlist, level)
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/plugins/processing/tools/vector.py", line 47, in <module>
			    from processing.tools import dataobjects, spatialite, postgis
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/qgis/utils.py", line 607, in _import
			    mod = _builtin_import(name, globals, locals, fromlist, level)
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/plugins/processing/tools/spatialite.py", line 29, in <module>
			    from pyspatialite import dbapi2 as sqlite
			  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/qgis/utils.py", line 607, in _import
			    mod = _builtin_import(name, globals, locals, fromlist, level)
			  File "/Library/Frameworks/SQLite3.framework/Versions/C/Python/2.7/pyspatialite/dbapi2.py", line 86, in <module>
			    register_adapters_and_converters()
			
```

**Comment by mnboos** [comment]
I think I know the reason: the map scale is neither set nor being updated in/by qgis.

Could you have a look at the scale in QGIS, right after loading zoom-level 0? Mine is set to 0 and won't even change.

See how the scale clearly not matches the actual zoom:
![image](https://user-images.githubusercontent.com/22575883/35097744-b12601a4-fc51-11e7-8f01-52889ee4ae1b.png)

**Comment by mnboos** [comment]
Fixed with commit 484eada.

Will be included in the next release v1.7.1
